myst_nb/transform.py

```diff
@@ -63,6 +62,136 @@ def apply(self):
+    """
+    import docutils
+    import os
+    from docutils.nodes import math_block
+    import nbconvert
+
+    to_add = []
+    for _, output in enumerate(cell.get("outputs", [])):
+        output_type = output["output_type"]
+        if output_type == "stream":
+            if output["name"] == "stderr":
+                if not write_stderr:
+                    continue
+                else:
+                    # Output a container with an unhighlighted literal block for
+                    # `stderr` messages.
+                    #
+                    # Adds a "stderr" class that can be customized by the user for both
+                    # the container and the literal_block.
+                    #
+                    # Not setting "rawsource" disables Pygment hightlighting, which
+                    # would otherwise add a <div class="highlight">.
+
+                    container = docutils.nodes.container(classes=["stderr"])
+                    container.append(
+                        docutils.nodes.literal_block(
+                            text=output["text"],
+                            rawsource="",  # disables Pygment highlighting
+                            language="none",
+                            classes=["stderr"],
+                        )
+                    )
+                    to_add.append(container)
+            else:
+                to_add.append(
+                    docutils.nodes.literal_block(
+                        text=output["text"],
+                        rawsource=output["text"],
+                        language="none",
+                        classes=["output", "stream"],
+                    )
+                )
+        elif output_type == "error":
+            traceback = "\n".join(output["traceback"])
+            text = nbconvert.filters.strip_ansi(traceback)
+            to_add.append(
+                docutils.nodes.literal_block(
+                    text=text,
+                    rawsource=text,
+                    language="ipythontb",
+                    classes=["output", "traceback"],
+                )
+            )
+        elif output_type in ("display_data", "execute_result"):
+            try:
+                # First mime_type by priority that occurs in output.
+                mime_type = next(x for x in data_priority if x in output["data"])
+            except StopIteration:
+                continue
+            data = output["data"][mime_type]
+            if mime_type.startswith("image"):
+                # Sphinx treats absolute paths as being rooted at the source
+                # directory, so make a relative path, which Sphinx treats
+                # as being relative to the current working directory.
+                filename = os.path.basename(output.metadata["filenames"][mime_type])
+
+                # checks if path in cell has a subpath inside of dir
+                filedir = os.path.dirname(output.metadata["filenames"][mime_type])
```

Comment:
these five lines

Reply:
It seems reasonable to me @akhmerov what do you think?

Reply:
Seems reasonable to me as well, except I don't quite understand the magical `filedir.split(dir)`. @AakashGfude can you elaborate more? Under which conditions is `len(subpaths) > 1`? Is `subpaths` ever empty?

Reply:
@akhmerov sure.  So, the below code will help in explaining 
![image](https://user-images.githubusercontent.com/6542997/80164518-8c57ec00-861c-11ea-8018-e94894fba41c.png)


As you can see, if `subpaths` exist, they are never empty. They have a size greater than one if the path passed as argument is a subpath.